solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,17 +19,33 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
+
     }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(@NonNull String filename,@NonNull Savable gameState) {
+            try (FileOutputStream fos = new FileOutputStream(filesDirectory + filename);
+             ObjectOutputStream out = new ObjectOutputStream(fos)) {
+            out.writeObject(gameState);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public Savable loadGame(@NonNull String filename) throws SaveGameException {
+
+        MapState mapState = null;
+
+            try (FileInputStream fis = new FileInputStream(filesDirectory + filename);
+             ObjectInputStream inputStream = new ObjectInputStream(fis)) {
+            mapState = (MapState) inputStream.readObject();
+        } catch (IOException e) {
+            throw new SaveGameException("I/O operation has been failed", SaveGameException.Type.IO, mapState);
+        } catch (ClassNotFoundException e) {
+            throw new SaveGameException("Class cast operation has been failed", SaveGameException.Type.SYSTEM, mapState);
```

Comment:
Line is longer than 120 characters (found 121).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1478167230>*